### DISCUSSION TITLE
Identify the devices ( Andriod or IOS)

### DISCUSSION THREAD

**Discussion by apppdevv2015** [owntracks/android#1770] [Q&A] [answered]
Hi Team,

Is there any way to identify the devices like ( Andriod or IOS) while sharing the location? Who used which device?

**Comment by jpmens** [comment]
This is currently possible if you know the data well. For instance Android currently adds an `_id` field, which iOS doesn't.

There should not really be a need for such an identification. Can you elaborate why you'd need that?

**Reply by apppdevv2015** [reply]
we are pushing the custom regions to devices so both have different configurations like CMD needs in IOS
example IOS : owntracks/testUser/myphone/cmd
Andriod: owntracks/testUser/waypoints while importing the waypoints

**Reply by jpmens** [reply]
I have explained at least once before to you, that you can use the same command (different topic obviously) on Android.

**Reply by ckrey** [reply]
If the latter is true (android) it is a bug

**Reply by jpmens** [reply]
I meant and originally said "different topic because the other device is subscribed to a different topic"

**Reply by ckrey** [reply]
both topics are `owntracks/<user>/<device>/cmd`

**Reply by apppdevv2015** [reply]
I know @jpmens and it was great help. I tried with ImportWaypoints feature and Its working however while setting below on Android phone and code  the below is not working
owntracks/testUser/myphone/cmd
owntracks/testUser/myphone/waypoint

**Reply by jpmens** [reply]
I said to your before already, and you didn't respond, that you did not show me proof that it is not working.

**Reply by apppdevv2015** [reply]
Importwaypoints is working fine for while using the Andriod device. I want to send the waypoint through Javascript code

**Comment by jpmens** [comment]
Closing this for boredom until OP demonstrates that Android CMD for waypoints import doesn't work.

**Comment by jpmens** [comment]
Please create an issue if you can show Android CMD import doesn't work.

**Reply by apppdevv2015** [reply]
ImportWaypoints is working fine for while using the Andriod device. I want to send the waypoint through Javascript code

**Comment by jpmens** [comment]
You are not listening to what we are saying.

**Reply by apppdevv2015** [reply]
Let me share the attachments

**Comment by apppdevv2015** [comment]
![1722258063180](https://github.com/user-attachments/assets/c7477b56-1f3b-4535-8382-e305aeff9e41)
[wp-andriod.json](https://github.com/user-attachments/files/16413666/wp-andriod.json)

I have attached the settings of Andriod

const waypointsFile = 'wp-andriod.json';
const waypoints = fs.readFileSync(waypointsFile, 'utf-8');
const topic = 'owntracks/testUser/oneplus9r/cmd'
client.on('connect', () => {
    console.log('Connected to MQTT broker');
    
    // Publish the waypoints
    client.publish(topic, waypoints, { qos: 1 }, (err) => {
      if (err) {
        console.error('Failed to publish waypoints:', err);
      } else {
        console.log('Waypoints published successfully');
      }
      client.end();
    });
  });

**Reply by jpmens** [reply]
that is not the way you sent the data to iOS.

How often must I repeat: THE. SAME. WAY.   THE. SAME. DATA FORMAT

**Reply by apppdevv2015** [reply]
Yes, this is the same way and its working fine for IOS

**Reply by jpmens** [reply]
Does this look familar to you?

```json
{
  "_type": "cmd",
  "action": "setWaypoints",
  "waypoints": {
    "_type": "waypoints",
    "waypoints": [
      {
        "_type": "waypoint",
        "tst": 1361636517,
        "rid": "blauw-utrecht-nl",
        "desc": "Restaurant Blauw",
        "rad": 50,
        "lat": 52.087822392366,
        "lon":  5.119438515394096
      }
    ]
  }
}
```

**Reply by apppdevv2015** [reply]
Yes, its working fine for IOS

**Reply by apppdevv2015** [reply]
When i tried to use the same in Android, its not working

**Reply by jpmens** [reply]
would you now PLEASE compare the data I have shown which you transmitted successfully to iOS and the data you have shown from your `wp-andriod.json` file?

**Reply by jpmens** [reply, marked as answer]
and also follow the instructions from @ckrey

**Reply by apppdevv2015** [reply]
[
[wp-andriod.json](https://github.com/user-attachments/files/16413790/wp-andriod.json)
]

**Reply by apppdevv2015** [reply]
@jpmens @ckrey ,

Thanks.. I uninstalled the app and installed again. Just quick one thing:

Can we push multiple regions at same time for same devices?

**Reply by jpmens** [reply]
Read the documentation Booklet on how to do that.

**Reply by apppdevv2015** [reply]
Yes, i read that,, It will overwrite the same day region

**Comment by ckrey** [comment]
`“sub“:false` means the device does not subscribe at all

I suggest you start from scratch with a clean install an only change the parameters you fully understand

**Reply by apppdevv2015** [reply]
Thank you guys, its working fine now.. I uninstalled the app and installed again

**Comment by apppdevv2015** [comment]
Hi @jpmens , @ckrey 

Yesterday we tested the waypoints in Android devices and it was working fine.. with same file of ios and Andriod however today its not working in any devices but working fine in IOS. We installed the app today in few mobiles
[wp-andriod.json](https://github.com/user-attachments/files/16423850/wp-andriod.json)

**Reply by jpmens** [reply]
who is "we" ?

**Reply by jpmens** [reply]
it's time you explain what "you" are trying to accomplish on how many mobile devices

**Reply by apppdevv2015** [reply]
Hi @jpmens,

My friends, we are team of 5 peoples here and tried to push the waypoints to the android devices. Yesterday, it was working fine but today not able to push anything

**Reply by jpmens** [reply]
is this a commercial enterprise?

**Reply by apppdevv2015** [reply]
No

**Reply by apppdevv2015** [reply]
Is there any issues in APP? for now, I am just trying my device only?
